app/src/main/java/com/example/plantsapp/presentation/ui/profile/ProfileFragment.kt

```diff
@@ -18,13 +21,21 @@ class ProfileFragment : Fragment(R.layout.fragment_profile) {
 
     private val binding: FragmentProfileBinding by viewBinding(FragmentProfileBinding::bind)
     private val viewModel: ProfileViewModel by viewModels()
+    private var isSignOutBtnVisible = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         with(viewModel) {
             user.observe(viewLifecycleOwner) { user ->
                 showUserProfile(user)
+                isSignOutBtnVisible = true
+                requireActivity().invalidateOptionsMenu()
```

Comment:
If this thing with observing from viewmodel will work, you can remove `requireActivity().invalidateOptionsMenu() `and  `private var isSignOutBtnVisible = false`